src/main/java/ch/uzh/ifi/hase/soprafs21/service/UserService.java

```diff
@@ -157,17 +164,21 @@ public String refreshToken(String refreshToken) {
     }
 
     public void updateRefreshTokenForUser(String newRefreshToken, String emailId) {
-
         Optional<User> optionalUser = userRepository.findByEmail(emailId);
-
         if(optionalUser.isPresent()) {
-
             User user = optionalUser.get();
             user.setToken(newRefreshToken);
             userRepository.saveAndFlush(user);
         }
+    }
 
-
-
+    public void logoutUser(Long userId) {
```

Comment:
This method should be Transactional, since it is CRUD operation

Reply:
Added